### DISCUSSION TITLE
Changelogs and update procedure

### DISCUSSION THREAD

**Discussion by xalex75** [raomin/ESPAltherma#220] [Q&A]
First, thank you again raomin, your job helped me a lot in finding the best settings to achieve a high confort and very low energy usage.

Then... probably it's just me not able to find them, but... are there step-by-step instructions on what should be changed in which files to update to a new version without risk of loosing settings? 
And where can I find the changelogs for each release, so I can evaluate if I need to update?

Thanks again.

**Comment by xalex75** [comment]
Would copying from old folder to new folder platformio.ini and setup.h files + def folder be enough?

**Reply by raomin** [reply]
Yes! 🙂

**Comment by Tardymo** [comment]
That is unless something changes in source code, making your backed-up setup.h  and def header file incompatible.

**Reply by xalex75** [reply]
> That is unless something changes in source code, making your backed-up setup.h and def header file incompatible.

And is there a way to check it that I didn't see?
Thanks

**Reply by Tardymo** [reply]
No easy way to do that. But unless there are some major changes done to this project, backed-up files should work just fine.
There is no change log, because there are no releases as such. Just one main branch in git. You can however go to commits section to see what has been changed and when. And then you can decide whether you need to update.
https://github.com/raomin/ESPAltherma/commits/main

**Reply by xalex75** [reply]
Thanks, I didn't find the commits page before.
That's more than enough. :-)

**Comment by d3zit** [comment]
Hi! 

Do we have a way of understanding which version of the local repository we have?
Also, is normal that source control in VScode is always showing the number of files edited/changed? It makes sense if you're the coder of course.
![image](https://user-images.githubusercontent.com/72941345/213159162-d840495a-0c4d-49b0-bcd7-7246ed85908b.png)

**Reply by Tardymo** [reply]
That's because you have to edit actual source files to configure ESPAltherma. This is not ideal and could be improved in future. Having modified uncommitted changes also makes harder to see when there are new commits in the repository and do an update. With no local changes, "Commit" button would show how many commits to pull from the origin repository. But I think that un-pulled change count can still be seen next to branch name. For example, here I have 30 commits to pull:
![image](https://user-images.githubusercontent.com/21990222/213256216-4333efcf-710b-446f-a561-b7b4dea6fd9d.png)

But, before you are able to pull, you need to first stash your changes, then do a pull, and then apply stash back to your project.
